docker-compose.yml

```diff
@@ -24,30 +26,39 @@ services:
       dockerfile: Dockerfile
     restart: on-failure
     ports:
-      - 8080:8080
+      - '8080:8080'
+    networks:
+      - zpi-app
 
-  ## THIS DOES NOT WORK, TO BE FIXED:
-  # frontend:
-  #   container_name: frontend
-  #   build: ./frontend
-  #   restart: on-failure
-  #   ports:
-  #     - 3000:3000
-  #   networks:
-  #       - full-stack-app
+  clap:
+    container_name: clap
+    build:
+      context: ./backend/clap
+      dockerfile: Dockerfile
+    restart: on-failure
+    ports:
+      - '8000:8000'
+    environment:
+      FRONTEND.ADDRESS: http://frontend:3000
+      GRPC.CLIENT.DATASTORE.ADDRESS: static://dstor:8080
+    networks:
+      - zpi-app
 
-  ## THIS DOES NOT WORK, TO BE FIXED:
-  # clap:
-  #   container_name: clap
-  #   build:
-  #     context: ./backend/clap
-  #     dockerfile: Dockerfile
-  #   restart: on-failure
-  #   ports:
-  #     - 8000:8000
-  #   networks:
-  #     - full-stack-app
+  # Correct if you need
+  frontend:
+    container_name: frontend
+    depends_on:
+      - clap
+    build: ./frontend
+    restart: on-failure
+    ports:
+      - '3000:3000'
+    networks:
+      - zpi-app
 
 volumes:
   db:
     driver: local
+
+networks:
```

Comment:
This network is not need and i removed it on purpose. When you do `docker compose` it automatically creates that. network. This just obfuscates it. Beside why should `frontend` or `clap` be in the same network as `db`  if they are not supposed to communicate? 

As i understand it without it you can refer to endpoint of `dstor` container as `dstor:8080`, just as dstor refers to database via `db:5432` instead of `localhost:8080` or `localhost:5432`

Reply:
You saw it and didn't care about security? Everyone can attack database because it's on public port, dstor as well. I am correcting it.